lib/config/expression.cpp

```diff
@@ -949,16 +949,26 @@ ExpressionResult ObjectExpression::DoEvaluate(ScriptFrame& frame, DebugHint *dhi
 	CHECK_RESULT(typeres);
 	Type::Ptr type = typeres.GetValue();
 
-	String name;
+	Value value;
 
 	if (m_Name) {
 		ExpressionResult nameres = m_Name->Evaluate(frame, dhint);
 		CHECK_RESULT(nameres);
 
-		name = nameres.GetValue();
+		value = nameres.GetValue();
+	}
+
+	if (!m_Abstract || !m_DefaultTmpl) {
+		if (value.IsEmpty()) {
+			BOOST_THROW_EXCEPTION(ScriptError("Template/object name must not be empty", m_DebugInfo));
+		}
+	}
+
+	if (!m_DefaultTmpl && !value.IsString()) {
```

Comment:
Any reason for a different condition here?

Reply:
What if a user creates a default template with an integer name and wants to import this manually? Anyway, you only mentioned that the creation of a default template with `template T /* null */` has still to work with this PR.

Reply:
Other templates must have a valid name like any other non-abstract objects.

Reply:
However, when it is checking for emptyness, it should only be checked for normal objects and non default templates. So that your statement of the referenced issue can be guaranteed.